src/abstraction-memory.c

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: LGPL-2.1-only
+/**
+ * Libcgroup abstraction layer for the memory controller
+ *
+ * Copyright (c) 2021-2025 Oracle and/or its affiliates.
+ * Author: Tom Hromatka <tom.hromatka@oracle.com>
+ */
+
+#include "abstraction-common.h"
+#include "abstraction-map.h"
+
+#include <libcgroup.h>
+#include <libcgroup-internal.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdio.h>
+
+#define V1_MAX_STR "9223372036854771712"
+#define V2_MAX_STR "max"
+
+#define LL_MAX 8192
+
+static int convert_limit_to_max(struct cgroup_controller * const dst_cgc,
+		const char * const in_setting, const char * const in_value,
+		const char * const out_setting)
+{
+	char max_line[LL_MAX] = {0};
+	int ret;
+
+	if (strlen(in_value) == 0) {
+		ret = cgroup_add_value_string(dst_cgc, out_setting, NULL);
+		if (ret)
+			goto out;
```

Comment:
Its coding style preference comment, you may want to remove the check for `ret` and goto branch. They are nop, as the next statement in both if `ret` checks will be `return ret` but without the extra branch and check.

Reply:
I think this is a good idea here.  It cleans it up a lot and the `out` tag can be entirely removed.